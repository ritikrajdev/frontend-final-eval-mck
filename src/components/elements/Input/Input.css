.custom-input {
  min-width: 250px;
  position: relative;
}

.custom-input label {
  font-weight: 600;
  color: var(--light-gray);
}

.custom-input input {
  box-sizing: border-box;
  padding: 8px;
  border-radius: 8px;
  border: none;
  margin-top: 8px;
  filter: drop-shadow(0px 0px 4px var(--dark-purple));
  width: 100%;
}
