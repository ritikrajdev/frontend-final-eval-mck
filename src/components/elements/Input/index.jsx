import PropTypes from 'prop-types';
import React from 'react';

import './Input.css';

export default function Input({
  label,
  type,
  onChange,
  name,
  placeholder,
  value,
  disabled,
}) {
  return (
    <div className='custom-input'>
      <label>{label}</label>
      <br />
      <input
        type={type}
        onChange={onChange}
        name={name}
        placeholder={placeholder}
        value={value}
        disabled={disabled}
      />
    </div>
  );
}

Input.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
  onChange: PropTypes.func,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  disabled: PropTypes.bool,
};
