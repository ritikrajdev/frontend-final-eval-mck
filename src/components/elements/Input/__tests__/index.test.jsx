import { render } from '@testing-library/react';
import React from 'react';
import Input from '..';

describe('Input', () => {
  it('should render with all the inputs', () => {
    const view = render(
      <Input
        label='Email'
        name='email'
        placeholder='Enter your email'
        type='email'
        onChange={() => {}}
        value='yo'
      />
    );

    expect(view.getByText('Email')).toBeTruthy();
    expect(view.asFragment()).toMatchSnapshot();
  });
});
