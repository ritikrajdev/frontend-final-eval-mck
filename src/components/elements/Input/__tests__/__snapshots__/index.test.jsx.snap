// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render with all the inputs 1`] = `
<DocumentFragment>
  <div
    class="custom-input"
  >
    <label>
      Email
    </label>
    <br />
    <input
      name="email"
      placeholder="Enter your email"
      type="email"
      value="yo"
    />
  </div>
</DocumentFragment>
`;
